contracts/governance/BaseWeightedMultisig.sol

```diff
@@ -0,0 +1,185 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import { IBaseWeightedMultisig } from '../interfaces/IBaseWeightedMultisig.sol';
+import { ECDSA } from '../libs/ECDSA.sol';
+
+abstract contract BaseWeightedMultisig is IBaseWeightedMultisig {
```

Comment:
Maybe we should have this as a base contract for the GMP Gateway instead of external module. We had code size issues with the original gateway and had to make it a module. 

In order for it to be a module it also needs to be an Ownable and have a little different setup